main_test.go

```diff
@@ -194,3 +203,53 @@ func registeredTestWslInstances() ([]wsl.Distro, error) {
 
 	return distros, nil
 }
+
+// defaultDistro gets the default distro's name via wsl.exe to bypass wsl.DefaultDistro in order to
+// better decouple tests.
+func defaultDistro() (string, error) {
+	out, err := exec.Command("powershell.exe", "-Command", "$env:WSL_UTF8=1; wsl.exe --list --verbose").CombinedOutput()
+	if strings.Contains(string(out), "Windows Subsystem for Linux has no installed distributions.") {
```

Comment:
shouldn’t we check `err != nil` before getting `out`? We can’t ensure the first parameter has valid content if it errored out.

Reply:
Well that message will only be printed when an error is thrown. I will make it safer by ensuring the error is `exec.ExitError` before reading the output.